mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -375,4 +377,31 @@ void CMomentumPlayer::CalculateAverageStats()
 
     // think once per 0.1 second interval so we avoid making the totals extremely large
     SetNextThink(gpGlobals->curtime + AVERAGE_STATS_INTERVAL, "THINK_AVERAGE_STATS");
-}
\ No newline at end of file
+}
+//this starts the timer when the player jumps inside the start zone, stopping
+//their timer again when they land and are not jumping.
+void CMomentumPlayer::StartTimerBhopOnly()
+{
+    ConVarRef gm("mom_gamemode");
+    CTriggerTimerStart *startTrigger = g_Timer.GetStartTrigger();
+    bool bhopGameMode = (gm.GetInt() == MOMGM_BHOP || gm.GetInt() == MOMGM_SCROLL);
+    //limit prespeed in the same way even if we aren't on a bhop gamemode.
+    if (m_bInsideStartZone)
+    {
+        if (GetGroundEntity() == NULL)
+            m_nTicksInAir++;
+        else
+            m_nTicksInAir = 0;
+        
+        if (bhopGameMode ? ((!g_Timer.IsRunning() && m_nTicksInAir > MAX_AIRTIME_TICKS) || g_Timer.IsPracticeMode(this)) : g_Timer.IsPracticeMode(this))
+        {
+            Vector velocity = GetLocalVelocity();
+            if (velocity.Length2D() > startTrigger->GetPunishSpeed())
```

Comment:
Would be more efficient to give the punish velocity squared and just use velocity.Length2DSqr()